Amend .net email authenticator guide to reflect bug fix (#3365)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/authenticators-okta-email/main/aspnet/integratechallengemagiclink.md

> 

Reviewer: Feedback on the new diagram:

1. In Step 4, "Returns status AwaitingChallenge..." should be "Return status AwaitingChallenge..."
2. In Step 4, make title consistent with text. Is it "4. Display a list of available ..." or 4. Display a list of possible ..."
3. In Step 6, the response arrows are pointing the wrong way.
4. Send email lto user's email address doesn't have a corresponding step in the content. Either add an item in the content or remove it from the diagram.
5. "7. Click the email magic link" text is not consistent with the corresponding header content
6. "8. Handle redirect in your app" text is not consistent with the corresponding header content
7. "Step 9. Complete challenge and sign user in" is missing from the content. Either remove it from the diagram or add it as a step in the content.
8. In step 8, "Passing in ..." should be "passing in ..."
9. In step 8, why is OPT bold in "Validate OTP". To be consistent with bolding the verbs just Validate should be bolded.

Author: I have fixed all issues.

Reviewer: For step 8 "Passing in" should be "passing in" since it's the same sentence.

</div>

### 5. Submit the email authenticator

> When the user selects the Email Authenticator and clicks **Submit**, the form posts back to the `SelectAuthenticatorAsync` method. This checks whether the user is in Challenge or Enrollment Flow.

Reviewer: What is "This" referring to in "This checks ..."?

Author: @robertnicolo-okta "This" refers to the SelectAuthenticatorAsync method. 

Changed text to "`SelectAuthenticatorAsync` method, which checks"

  (selectAuthenticatorOptions, (IIdxContext)Session["IdxContext"]);
```

> If the call is successful, OIE sends an email to the user and the returned `selectAuthenticatorResponse` object has an `AuthenticationStatus` of `AwaitingAuthenticatorVerification`. This indicates that OIE is waiting for the user to check their email and either click the magic link or enter the OTP.

Reviewer: Get we ask Susan and Van about the OIE usage here. I'm not quite sold on it and like to get their take on what to call the Okta code that sends the email.  I ask this because, all our guides should be moving to "OIE" or whatever term is decided to be consistent.

Author: @robertnicolo-okta Agreed - we can't keep leaving it at "Okta sends an email" or "Okta does x"

Author: Have changed OIE to Identity Engine as discussed elsewhere.

</section>
```

> ### 7. User clicks the email magic link

Reviewer: The diagram lists this as "7. Click the email magic link". Make this text consistent with the diagram.

When the user clicks the magic link, your org receives the request, gets the OTP and state parameters, and forwards the request with these parameters to your application. The org combines the Callback URI that you defined in [Update configurations](#update-configurations) with the OTP and state parameters to produce a final callback URL for the user. For example, `https://localhost:44314/magiclink/callback?otp=726009&state=1b34371af02dd31d2bc4c48a3607cd32`

### 8. Handle the magic link redirect in your app

> Create a callback handler method that checks the user has opened the magic link in the same browser and on the same device they used for the previous steps of the challenge flow. If this is true, take the `otp` parameter in the query string and pass it as a parameter to `IdxClient.VerifyAuthenticatorAsync()`.

Reviewer 2: ```suggestion
Create a callback handler method that checks if the user has opened the magic link in the same browser and on the same device that they used for the previous steps of the challenge flow. If this is true, take the `otp` parameter in the query string and pass it as a parameter to `IdxClient.VerifyAuthenticatorAsync()`.
```

When the user clicks the magic link, your org receives the request, gets the OTP and state parameters, and forwards the request with these parameters to your application. The org combines the Callback URI that you defined in [Update configurations](#update-configurations) with the OTP and state parameters to produce a final callback URL for the user. For example, `https://localhost:44314/magiclink/callback?otp=726009&state=1b34371af02dd31d2bc4c48a3607cd32`

> ### 8. Handle the magic link redirect in your app

Reviewer: The diagram lists this as "8. Handle redirect in your app". Make this text consistent with the diagram.

    VerifyAuthenticatorAsync(verifyAuthenticatorOptions, idxContext);
```

> If the user has not opened the magic link in the same browser, the `otp` value isn't valid, or an `AuthenticationStatus` is returned that is not handled by your case statement, you should advise the user to return to the original tab in the browser and enter the `otp` value there to proceed.

Reviewer 2: ```suggestion
If the user hasn't opened the magic link in the same browser, the `otp` value isn't valid, or an `AuthenticationStatus` is returned that isn't handled by your case statement, you should advise the user to return to the original tab in the browser and enter the `otp` value there to proceed.
```
Suggestion: This sentence was difficult to follow - had to read a few times. Maybe try this construction?

For the following conditions, advise the user to return to the original tab in the browser and enter the `otp` value to proceed:

- User hasn't opened the magic link in the same browser
- The `otp` value isn't valid
- An `AuthenticationStatus` is returned that isn't handled by your case statement

Author: Fixed.

If the `otp` value is valid, the `AuthenticationStatus` property of the `AuthenticationResponse` object returned by `VerifyAuthenticatorAsync` is `Success`. In this case, call `AuthenticationHelper.GetIdentityFromTokenResponseAsync` to retrieve the OIDC claims information about the user and pass them into your application. The user has now signed in.

```csharp
switch (authnResponse.AuthenticationStatus)
{
   // other case statements

   case AuthenticationStatus.Success:
   ClaimsIdentity identity = await AuthenticationHelper.
      GetIdentityFromTokenResponseAsync(_idxClient.Configuration, authnResponse.TokenInfo);
   _authenticationManager.SignIn(new AuthenticationProperties(), identity);
   return RedirectToAction("Index", "Home");
}
```

If the `otp` value isn't valid or an `AuthenticationStatus` is returned that is not handled by your case statement, you should advise the user to return to the original tab in the browser and enter the `otp` value there to proceed.

```csharp
return View(new MagicLinkCallbackModel {
   Message = $"Please enter the OTP '{otp}' in the original browser tab to finish the flow."
});

> ```

Reviewer: "Step 9. Complete challenge and sign user in" is missing from the content. Either remove it from the diagram or add it as a step in the content.
